### ISSUE TITLE
Bug with manifest storage and relative URLs

### ISSUE THREAD

**Issue by shauns** [etianen/django-s3-storage#40]
Consider Django's `base.css`: https://github.com/django/django/blob/1.10.5/django/contrib/admin/static/admin/css/base.css#L356

It uses a relative reference to another resource:
```
background: url(../img/sorting-icons.svg) 0 0 no-repeat;
```

When it comes to post-processing, we end up calling `.exists('../img/sorting-icons.svg')` method in this library. `_get_key_name` returns `shaun-test/admin/css/../img/sorting-icons.svg`, *not* `shaun-test/admin/img/sorting-icons.svg` as expected. This looks to be a regression caused by https://github.com/etianen/django-s3-storage/commit/893608027e8a78dc9b57f783d0c21541cbd2dad2 

I'll try and get a test case/fix together.

**Comment by etianen** [comment]
Hmm... is this your own code causing the issue, or is it core Django code?

The reason I ask, is that there's no way for django-s3-storage to resolve this relative URL correctly, since it needs to know the URL of the CSS file that embeds the relative URL. I'd have thought that Django's post-processors would resolve the relative URL before passing it onto the storage.

**Comment by shauns** [comment]
As far as I can recreate, the core Django code -- I'll try to make a failing test case.

**Comment by shauns** [comment]
Failing test above: Django calls `exists('admin/css/../img/sorting-icons.svg')`

**Comment by etianen** [comment]
Thanks for reporting this and looking into a test case and fix. Awesomely helpful! 😀 

I've modified #40. Now all S3 keys are normalized to use unix-style path separators, and resolve relative paths. This is because I can see this bug rearing it's ugly head for all the different storage methods, so it may as well be fixed now.

As a result, this is an edge-case breaking change, so I've bumped the version to 0.11.0.
